scripts/write_nz_building_outline.py

```diff
@@ -0,0 +1,209 @@
+    """Create and encode geo metadata dict.
+
+    Parameters
+    ----------
+    df : GeoDataFrame
+
+    Returns
+    -------
+    dict
+    """
+
+    # Construct metadata for each geometry
+    column_metadata = {}
+    for col, geometry_array in geometry_columns.items():
+        geometry_type = get_geometry_type(geometry_array)
+        bbox = list(pygeos.total_bounds(geometry_array))
+
+        series = df[col]
+        column_metadata[col] = {
+            "encoding": "WKB",
+            "geometry_type": geometry_type,
+            "crs": series.crs.to_wkt(version="WKT2_2019") if series.crs else None,
+            # We don't specify orientation for now
+            # "orientation"
+            "edges": "planar",
+            "bbox": bbox,
+            # I don't know how to get the epoch from a pyproj CRS, and if it's relevant
+            # here
+            # "epoch":
+        }
+
+    return {
+        "version": GEOPARQUET_VERSION,
+        "primary_column": df._geometry_column_name,
+        "columns": column_metadata,
+        # "creator": {"library": "geopandas", "version": geopandas.__version__},
+    }
+
+
+def get_geometry_type(pygeos_geoms: PygeosGeometryArray) -> List[str]:
+    type_ids = pygeos.get_type_id(pygeos_geoms)
+    unique_type_ids = set(type_ids)
+
+    geom_type_names: List[str] = []
+    for type_id in unique_type_ids:
+        geom_type_names.append(GeometryType(type_id).name)
+
+    return geom_type_names
+
+
+def encode_metadata(metadata: Dict) -> bytes:
+    """Encode metadata dict to UTF-8 JSON string
+
+    Parameters
+    ----------
+    metadata : dict
+
+    Returns
+    -------
+    UTF-8 encoded JSON string
+    """
+    return json.dumps(metadata).encode("utf-8")
+
+
+def cast_dtypes(df: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    """
+    Note: This is specific to the nz-building-outlines data
+    See reference here:
+    https://nz-buildings.readthedocs.io/en/latest/published_data.html#table-nz-building-outlines
+    """
+    # Double checks
+    assert df["building_id"].min() >= np.iinfo(np.int32).min
+    assert df["building_id"].max() <= np.iinfo(np.int32).max
+    df["building_id"] = df["building_id"].astype(np.int32)
+
+    assert df["capture_source_id"].min() >= np.iinfo(np.int32).min
+    assert df["capture_source_id"].max() <= np.iinfo(np.int32).max
+    df["capture_source_id"] = df["capture_source_id"].astype(np.int32)
+
+    for date_col in ["capture_source_from", "capture_source_to", "last_modified"]:
+        df[date_col] = pd.to_datetime(df[date_col], format="%Y-%m-%d")
+
+    return df
+
+
+def geopandas_to_arrow(df: gpd.GeoDataFrame) -> pa.Table:
+    geometry_columns = parse_to_pygeos(df)
+    geo_metadata = _create_metadata(df, geometry_columns)
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message=".*does not contain geometry.*")
+        for col, geometry_array in geometry_columns.items():
+            df[col] = pygeos.to_wkb(geometry_array)
```

Comment:
```suggestion
    df = pd.DataFrame(df)
    for col, geometry_array in geometry_columns.items():
        df[col] = pygeos.to_wkb(geometry_array)
```

Alternative to avoid the warning

Reply:
Thanks!